src/node_worker.cc

```diff
@@ -918,6 +924,56 @@ void GetEnvMessagePort(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+void SetMainPort(const FunctionCallbackInfo<Value>& args) {
+  if (args.Length() < 1 || args[0]->IsNullOrUndefined()) {
+    main_port = nullptr;
+    return;
+  }
+
+  MessagePort* port;
+  ASSIGN_OR_RETURN_UNWRAP(&port, args[0]);
+
+  main_port = port;
+}
+
+void SendToWorker(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Local<Context> context = env->context();
+  v8::Isolate* isolate = args.GetIsolate();
+
+  if (args.Length() < 2) {
+    return THROW_ERR_MISSING_ARGS(env, "Not enough arguments to "
+                                       "SendToWorker");
+  }
+
+  CHECK(args[0]->IsUint32());
+  auto tid = args[0].As<v8::Uint32>()->Value();
+
+  MessagePort* port;
+
+  if (tid == 0) {
+    port = main_port;
+  } else {
+    auto it = workers.find(tid);
+    if (it == workers.end()) {
+      isolate->ThrowException(
+        ERR_WORKER_INVALID_ID(isolate, "Invalid worker id"));
+      return;
+    }
+
+    port = it->second->parent_port();
+  }
+
+  TransferList transfer_list;
+  if (args.Length() > 2) {
+    if (!GetTransferList(env, context, args[2], &transfer_list)) {
+      return;
+    }
+  }
+
+  port->PostMessage(env, context, args[1], transfer_list);
```

Comment:
Could we reach this line and have `nullptr` for `port`?

Reply:
Done